bats/tests/containers/allowed-images.bats

```diff
@@ -10,6 +10,7 @@ RD_USE_IMAGE_ALLOW_LIST=true
 
 @test 'update the list of patterns first time' {
     update_allowed_patterns true '"nginx", "busybox", "python"'
+    wait_for_container_engine
```

Comment:
Do we need the wait after other calls to `update_allowed_patterns`? For example, after we drop python. (We immediately run `ctrctl rmi` after, so if the engine is down that will fall over.)

If yes, it may make sense to bake that into `update_allowed_patterns`.

Reply:
I believe we only need to wait when `containerEngine.allowedImages.enabled` changes, because that needs a restart of the container engine to add/remove the proxy definition.

Changes to the patterns should just reload openresty, but does not require any proxy settings changes.

I belive that is how the code works right now, but it seems in need of refactoring.